tests/functional/test-lessee-id.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+project_id=75681640118a4890b3da0d106eae8af7 # test1 uuid
+tmpfile=$(mktemp ./leaseXXXXXX)
+nodefile=$(mktemp /tmp/nodes/XXXX)
+errfile=$(mktemp ./errXXXXXX)
+node_uuid=$(echo $nodefile | sed 's/\/tmp\/nodes\///')
+start=$(date +%Y-%m-%d)
+end=$(date -d "+5 days" +%Y-%m-%d)
+lessee='test1-subproject'
+
+# Setup script, contains offer create test
+
+./setup.sh $node_uuid $project_id $tmpfile $start $end $lessee
+
+. $tmpfile
+
+# The uuid variable will be overwritten, so we save it here
+test_offer_uuid=$uuid
+# Run cleanup if the script exits, contains offer delete test
+trap "./cleanup.sh $test_offer_uuid; rm -f $tmpfile $errfile $nodefile" EXIT 
```

Comment:
Is it possible to delete these files as part of the cleanup script?

Reply:
Yeah, I could do that. I didn't want to bother writing it because it's actually more code than this to pass the filename variables in there, but if you think it'd be cleaner that way I'm happy to.

Reply:
Okay, looking at the code more closely - I think tmpfile and errfile could stay within the test script, since the scripts actively use the file. I'm wondering if you could move the temp node creation into the setup script, set the node_uuid there, and then delete the node file in the cleanup script without ever passing the nodefile out.

Reply:
Maybe, but with the way mktemp works, the node id is defined as part of the node creation. If I had it in setup.sh, I would have to pass it out somehow.

Reply:
But aren't you passing out a bunch of information from setup.sh anyway? Is it possible to add the node id there?

Reply:
No, setup.sh just edits the tmpfile with the output from the create offer command. I could append the node_id there though I guess.

Reply:
Yep, let's do that then! And if you do that - maybe add the tmpfile and errfile as well.

Reply:
I can't add tmpfile, since that's how I'm passing the variables, so the test script needs to know where it is. I can add errfile and nodefile though.